qiskit/transpiler/passes/mapping/mapper.py

```diff
@@ -45,6 +47,10 @@ def __call__(self, circuit: DAGCircuit, current_mapping: Mapping[Reg, ArchNode])
             -> Mapping[Reg, ArchNode]:
         return self.map(circuit, current_mapping)
 
+    def run(self, dag: DAGCircuit) -> None:
+        new_mapping = self.map(circuit=dag, current_mapping=self.property_set['layout'])
+        self.property_set['new_mapping'] = Layout(input_dict=new_mapping)
```

Comment:
I'm not quite sure where to put the new mapping that is found by the mapper. The new mapping is a partial permutation so it cannot be used as the new layout before qubits are routed.

Reply:
@eddieschoute Sorry for not responding for a long time. 
I think you should create a new class (say ARCTSwap) inheriting TransformationPass with integrating your mapper and swapper. For that, I think my PR would be helpful:
https://github.com/Qiskit/qiskit-terra/blob/9ec90d18447d3932e0ed564e285554cadc5ee1e0/qiskit/transpiler/passes/mapping/flexlayer_swap.py
Leaving the arct_swap.py in passes/mapping folder and move all the other files to a subfolder (e.g. passes/mapping/algorithm) would be good to keep the passes/mapping folder simple.
Please ask me if you are not clear about what I'm suggesting.

Reply:
If I'm understanding you correctly, I probably don't agree with this approach. It would turn choosing classes into a string flag in a constructor. I think the point of the Passes is also that one can just mix and match a Passes quite easily instead of having reasonably opaque string flags.

Reply:
(As a side note, can we move this discussion to the main discussion page instead of a comment on a line?)